collections/ansible_collections/cloudkit/service/roles/manage_agents/tasks/import_agents.yaml

```diff
@@ -0,0 +1,135 @@
+    tasks_from: provision_nodes
+  vars:
+    provisioning_network_name: "{{ manage_agents_provisioning_network_name }}"
+    node_info_list: "{{ filtered_node_info }}"
+    wait: true
+
+- name: Get filtered agent names
+  ansible.builtin.include_role:
+    name: massopencloud.esi.node
+    tasks_from: get_agent_names
+  vars:
+    node_info_list: "{{ filtered_node_info }}"
+# registers: agent_names
+# registers: agent_hostnames
+
+- name: Display the node names that were imported as agents
+  ansible.builtin.debug:
+    msg: "Imported {{ node_info.name }} as agent {{ agent_names[node_idx] }}"
+  loop: "{{ filtered_node_info }}"
+  loop_control:
+    loop_var: node_info
+    index_var: node_idx
+    label: "{{ node_idx }}"
+
+- name: Extract topology from all node names
+  ansible.builtin.set_fact:
+    agent_topology_labels_list: >-
+      {{ agent_topology_labels_list | default([]) + [node_info.name | massopencloud.esi.extract_esi_location] }}
+  loop: "{{ filtered_node_info }}"
+  loop_control:
+    loop_var: node_info
+    label: "{{ node_info.name }}"
+
+- name: Create topology labels for all nodes
+  ansible.builtin.set_fact:
+    agent_topology_labels_list: >-
+      {{ agent_topology_labels_list | default([]) + [topology_labels] }}
+  vars:
+    topology_labels: |-
+      {% filter from_yaml %}
+      {% for label, val in agent_topology.items() %}
+        {{ manage_agents_topology_label_namespace }}/{{ label }}: "{{ val }}"
+      {% endfor %}
+      {% endfilter %}
+  loop: "{{ agent_topology_labels_list }}"
+  loop_control:
+    loop_var: agent_topology
+
+- name: Create resource class labels for all nodes
+  ansible.builtin.set_fact:
+    agent_resource_class_labels_list: >-
+      {{ agent_resource_class_labels_list | default([]) + [{'esi.nerc.mghpcc.org/resource_class': node_info.resource_class}] }}
+  loop: "{{ filtered_node_info }}"
+  loop_control:
+    loop_var: node_info
+    label: "{{ node_info.name }}"
+
+- name: Get node topology labels
+  ansible.builtin.include_role:
+    name: massopencloud.esi.node
+    tasks_from: extract_node_location
+  vars:
+    node_info_list: "{{ filtered_node_info }}"
+# registers: topology_labels_list
+
+- name: Configure agent metadata for all nodes
+  kubernetes.core.k8s:
+    state: present
+    definition:
+      apiVersion: agent-install.openshift.io/v1beta1
+      kind: Agent
+      metadata:
+        name: "{{ agent_names[node_idx] }}"
+        namespace: "{{ manage_agents_namespace }}"
+        annotations: "{{ {manage_agents_node_uuid_annotation: filtered_node_info[node_idx].id} }}"
+        labels: >-
+          {{
+            (topology_labels_list[node_idx] |
+            from_yaml) |
+            combine(agent_resource_class_labels_list[node_idx])
+          }}
+      spec:
+        approved: true
+        hostname: "{{ filtered_node_info[node_idx].name | lower }}"
+  loop: "{{ range(agent_names | length) | list }}"
+  loop_control:
+    loop_var: node_idx
```

Comment:
I think I'd have a single esi role that encompasses the setting of labels on a list of agents (so a role that extracts the topology and does the other things before setting the labels). Apparently different bare metal management systems may end up with a different set of labels, and what we're doing here is likely very ESI specific.